samza-core/src/main/java/org/apache/samza/table/remote/RemoteReadWriteTable.java

```diff
@@ -69,33 +68,38 @@ public void init(SamzaContainerContext containerContext, TaskContext taskContext
     super.init(containerContext, taskContext);
     writeMetrics = new DefaultTableWriteMetrics(containerContext, taskContext, this, tableId);
     TableMetricsUtil tableMetricsUtil = new TableMetricsUtil(containerContext, taskContext, this, tableId);
-    putThrottleNs = tableMetricsUtil.newTimer("put-throttle-ns");
+    writeRateLimiter.setTimerMetric(tableMetricsUtil.newTimer("put-throttle-ns"));
   }
 
   /**
    * {@inheritDoc}
    */
   @Override
   public void put(K key, V value) {
+    try {
+      putAsync(key, value).get();
+    } catch (InterruptedException e) {
+      throw new SamzaException(e);
+    } catch (Exception e) {
+      throw (SamzaException) e.getCause();
```

Comment:
The cause is already a SamzaException so rethrowing it avoid creating deeply nested exceptions.

Reply:
I think a cast here appears dangerous since it relies that there will be _no_ run-time exceptions eg: NPE etc. thrown by getAll(). I 'm not sure if it's a safe assumption to make.

Reply:
But if we look at putAsync, we catch any kind of exceptions (exceptionally) and wrap it with a SamzaException. That's why I think the cast here is kind safe, right?

Reply:
the whole point is to avoid any such coupling.

Reply:
Will wrap it.